src/math/math_base.h

```diff
@@ -204,6 +216,19 @@ class LineBase2D
         return (distanceToLine < 0 ? -distanceToLine : distanceToLine);
     }
 
+    PointBase2D<_Type> projection( const PointBase2D<_Type> & point ) const
+    {
+        const _Type dotProduct = _direction.x * ( point.x - _position.x ) + _direction.y * ( point.y - _position.y );
+        const PointBase2D<_Type> offset( _direction.x * dotProduct, _direction.y * dotProduct );
+        return _position + offset;
+    }
+
+    PointBase2D<_Type> opposite( const PointBase2D<_Type> & point ) const
+    {
+        const PointBase2D<_Type> offset = point - projection( point );
+        return point - 2 * offset;
```

Comment:
Let's try this way:
```cpp
return 2 * projection( point ) - point;
```

which comes from this:
```
point - 2 * offset
point - 2 * ( point - projection( point ) )
point - 2 * point + 2 * projection( point )
-point + 2 * projection( point )
2 * projection( point ) - point
```

Such approach will improve an accuracy of calculations.

Reply:
You know, I tried this way. The problems still present.

I have another suggestion how to solve this problem but you could don't like it because it changes the `pwmath::isEqual()` method. Look at my last commit, please.  [c5fdd1ad](https://github.com/ihhub/penguinV/pull/577/commits/c5fdd1adcccdadfba1405beae5d1d197009c02ed)

Reply:
That might be because float rounds up to 7 digits. When we have little numeral (up to 100), 7 digits are 99.99999. And it's inside the accuracy of 1e-5. But if we have the point(-100, -999) and want to find opposition of line ((0, 999)(1, 999)), we'll get the point(-100, 1998). And 7 digits will be 1998.000 which are not enough to provide accuracy of 1e-5.

Reply:
No, it's incorrect. Float point variables are under IEEE 754. We loose an accuracy because of non-ideal computations. Please try with my changes and see the accuracy after.

Reply:
Could you please revert your changes for now and show me results after my suggested change. The problem lies in number of computations we do for opposite point. Less computations - more precise results.

Reply:
Let me find some info which would simplify the solution...